spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+
+describe UsersController do
+  let!(:first_user) { create(:user) }
+  let!(:user) { create(:user, name: 'My Name') }
+
+  describe 'GET #show' do
+    before :each do
+      get :show, id: user.id
+    end
+
+    it 'renders show template' do
+      expect(response).to render_template :show
+    end
+
+    it 'assigns the right value to @user' do
+      expect(assigns(:user)).to eq user
+    end
+
+    it 'assigns [] to @events, when user does not have any submissions' do
+      expect(assigns(:events)).to eq []
+    end
+
+    it 'assigns the correct value to @events, when the user has submissions' do
+      conference = create(:conference)
+      event = create(:event, state: 'confirmed', program: conference.program)
+      event.event_users = [create(:event_user, user: user, event_role: 'submitter')]
+
+      expect(assigns(:events)).to eq [event]
+    end
+  end
+
+  describe 'GET #edit' do
+    it 'assigns the right value to @user' do
+      sign_in user
+      get :edit, id: user.id
+      expect(assigns(:user)).to eq user
+    end
+  end
+
+  describe 'PATCH #update' do
+    context 'with valid attributes' do
+      before :each do
+        sign_in user
```

Comment:
It won't save us any lines, as I still need the individual before blocks. And since I have the before block anyway, it makes it more clear that I sign_in the user, and we don't have to search for it elsewhere (it is just 1 line).